### PR TITLE
Create redistribution archive for successful builds

### PR DESCRIPTION
**Pull request by hkaiser**
- flyby: export HPX_VERSION_STRING from HPXConfig.cmake

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -33,11 +35,28 @@ before_build:
...
+
+deploy:
+  provider: FTP
+  on:
+    branch: master
+  protocol: ftp
+  host: ftp.stellar-group.org
+  username: appveyor-ci@stellar-group.org
+  password:
+    secure: tsfxuDuvSntD/aqRcuxSvkU4lqQck85OI7e7nhDhIr4=
```
Comment (sithhell) [inline appveyor.yml]: Are you sure you want to store that at such a prominent spot?

File: appveyor.yml [hunk 2]
```diff
@@ -33,11 +35,28 @@ before_build:
...
   - cmd: cmake --build build --config %CONFIGURATION% --target examples -- /logger:"C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll" /verbosity:minimal /maxcpucount:2 /nologo
-#  - cmd: cmake --build build --config %CONFIGURATION% --target tests/regressions -- /logger:"C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll" /verbosity:minimal /maxcpucount:2 /nologo
 
 test_script:
   - cmd: |
       set PATH=%PATH%;C:\projects\vcpkg\installed\x64-windows\debug\bin
       python build\bin\hpxrun.py -l2 -t2 build\Debug\bin\hello_world.exe -- --hpx:bind=none
 
+artifacts:
+  - path: hpx-master-redist-$(MSVC_GENERATOR_SHORT)-$(configuration).7z
```
Comment (sithhell) [inline appveyor.yml]: What information does the configuration contain?

File: appveyor.yml (hunk 1 above)
Comment (Naios) [reply appveyor.yml]: I think this is the correct way of storing sensitive data in appveyor: https://www.appveyor.com/docs/how-to/secure-files/

Comment (hkaiser) [reply appveyor.yml]: @sithhell this is encrypted, so it's safe

File: appveyor.yml (hunk 2 above)
Comment (hkaiser) [reply appveyor.yml]: @sithhell 'Debug', 'Release', etc. It's essentially the `CMAKE_BUILT_TYPE`.

File: appveyor.yml (hunk 1 above)
Comment (sithhell) [reply appveyor.yml]: So I can't just copy the credentials and use them to store stuff on the ftp?

File: appveyor.yml (hunk 2 above)
Comment (sithhell) [reply appveyor.yml]: Ok. Does it make sense to at the commit hash?

File: appveyor.yml (hunk 1 above)
Comment (sithhell) [reply appveyor.yml]: Ok, got it.

File: appveyor.yml (hunk 2 above)
Comment (hkaiser) [reply appveyor.yml]: I intentionally didn't do that. All I need is the snapshot from the latest successful build. Similar to what we do in CircleCI.

**Pull request merged**